src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -128,6 +133,112 @@ internal partial class Binder
+                    Error(diagnostics, ErrorCode.ERR_NoConstructors, syntax, strippedType);
+                    hasErrors = true;
+                    break;
+
+                case TypeKind.Struct when strippedType.IsTupleType:
+                    Error(diagnostics, ErrorCode.ERR_NewWithTupleTypeSyntax, syntax, strippedType);
+                    hasErrors = true;
+                    break;
+
+                case TypeKind.Struct:
+                case TypeKind.Class:
+                    operand = BindClassCreationExpression(
+                        node,
+                        typeName: strippedType.Name,
+                        typeNode: node,
+                        type: (NamedTypeSymbol)strippedType,
+                        arguments,
+                        diagnostics,
+                        boundInitializerOpt,
+                        forTargetTypedNew: true);
+                    operand.WasCompilerGenerated = true;
+                    hasErrors = operand.HasErrors;
+                    break;
+
+                case TypeKind.TypeParameter:
+                    var typeParameter = (TypeParameterSymbol)strippedType;
+                    if (!typeParameter.IsInstantiable())
+                    {
+                        Error(diagnostics, ErrorCode.ERR_BadTargetTypeForNew, syntax, strippedType);
+                        hasErrors = true;
+                        break;
+                    }
+
+                    operand = BindTypeParameterCreationExpression(
+                        node,
+                        typeParameter,
+                        boundInitializerOpt,
+                        diagnostics);
+                    operand.WasCompilerGenerated = true;
+                    hasErrors = operand.HasErrors;
+                    break;
+
+                default:
+                    hasErrors = true;
+                    break;
+            }
+
+            arguments.Free();
+
+            var result = new BoundConversion(
+                syntax,
+                operand,
+                conversion: conversion,
+                @checked: false,
+                explicitCastInCode: isCast,
+                constantValueOpt: null, // A "target-typed new" would never produce a constant.
+                type: strippedType,
+                hasErrors: hasErrors)
+            { WasCompilerGenerated = true }; // The "implicit new" conversion can never be explicit in source.
+
+            if (destination.IsNullableType())
+            {
+                result = new BoundConversion(
+                    syntax,
+                    result,
+                    conversion: new Conversion(ConversionKind.ImplicitNullable, Conversion.IdentityUnderlying),
+                    @checked: false,
+                    explicitCastInCode: isCast,
+                    constantValueOpt: null,
+                    type: destination)
+                { WasCompilerGenerated = true };
+            }
```

Comment:
It can't be compiler generated and explicit in code at the same time.

Reply:
what values are expected for `explicitCastInCode` and `WasCompilerGenerated` in `(S?)new(x)`?

I guess the latter should be false.